lib/absinthe/phase/document/execution/resolution.ex

```diff
@@ -135,49 +206,102 @@ defmodule Absinthe.Phase.Document.Execution.Resolution do
     do_resolve_fields(fields, res_acc, info, source, [result | acc])
   end
 
+  # Normal :ok result
   defp build_result({:ok, result}, acc, bp_field, info, _) do
     full_type = Type.expand(bp_field.schema_node.type, info.schema)
 
     result
     |> to_result(bp_field, full_type)
     |> walk_result(acc, bp_field, full_type, info)
   end
-  defp build_result({:error, params} = other, acc, bp_field, info, source) when is_list(params) or is_map(params) do
-    case Keyword.split(Enum.to_list(params), [:message]) do
-      {[], _} -> result_format_error(other, bp_field, source)
-      {[message: msg], extra} ->
-        build_error_result(msg, extra, acc, bp_field, info)
-    end
+  # Error result; force wrap of single, single-value Keyword.t errors
+  defp build_result({:error, [{_, _}] = error_value} = err, acc, bp_field, info, source) do
+    build_error_result(err, [error_value], acc, bp_field, info, source)
   end
-  defp build_result({:error, msg}, acc, bp_field, info, _) do
-    build_error_result(msg, [], acc, bp_field, info)
+  # Error result; force wrap of single, multiple-value Keyword.t errors
+  defp build_result({:error, [{_, _} | _] = error_value} = err, acc, bp_field, info, source) do
```

Comment:
This would also match the case on line `218`, since `[{:foo, 1}]` is `[{:foo, 1} | []]`